agg-ast/schema_derivation/src/test.rs

```diff
@@ -77,6 +77,28 @@ mod get_schema_for_path {
         path = vec!["a".to_string(), "b".to_string(), "c".to_string()]
     );
 
+    test_schema_for_path!(
```

Comment:
I'm rocking with these tests. I think it might also be worth adding these in practice, perhaps not everywhere, but also in at least one likely operator. Perhaps we can add an $unwind test that relies on this?